.github/workflows/release.yml

```diff
@@ -147,16 +147,60 @@ jobs:
             aarch64-linux-gnu-strip \
             /target/${{ matrix.target }}/release/hx
 
-      - name: Build archive
+      - name: Build AppImage
         shell: bash
+        if: matrix.build == 'x86_64-linux'
         run: |
           mkdir dist
+
+          name=dev
+          if [[ $GITHUB_REF == refs/tags/* ]]; then
+            name=${GITHUB_REF:10}
+          fi
+
+          export VERSION="$name"
+          export ARCH=x86_64
+          export APP=helix
+          export OUTPUT="helix-$VERSION-$ARCH.AppImage"
+          export UPDATE_INFORMATION="gh-releases-zsync|$GITHUB_REPOSITORY_OWNER|helix|latest|$APP-*-$ARCH.AppImage.zsync"
+
+          mkdir -p "$APP.AppDir"/usr/{bin,lib/helix}
+
+          cp "target/${{ matrix.target }}/release/hx" "$APP.AppDir/usr/bin/hx"
+          rm -rf runtime/grammars/sources
+          cp -r runtime "$APP.AppDir/usr/lib/helix/runtime"
+
+          cat << 'EOF' > "$APP.AppDir/AppRun"
+          #!/bin/sh
+
+          APPDIR="$(dirname "$(readlink -f "${0}")")"
+          HELIX_RUNTIME="$APPDIR/usr/lib/helix/runtime" exec "$APPDIR/usr/bin/hx" "$@"
+          EOF
+          chmod 755 "$APP.AppDir/AppRun"
+
+          curl -Lo linuxdeploy-x86_64.AppImage \
+              https://github.com/linuxdeploy/linuxdeploy/releases/download/continuous/linuxdeploy-x86_64.AppImage
+          chmod +x linuxdeploy-x86_64.AppImage
+
+          ./linuxdeploy-x86_64.AppImage \
+              --appdir "$APP.AppDir" -d contrib/Helix.desktop \
+              -i contrib/helix.png --output appimage
+
+          mv "$APP-$VERSION-$ARCH.AppImage" \
+              "$APP-$VERSION-$ARCH.AppImage.zsync" dist
```

Comment:
This should build a separate archive. I don't want to bundle it together with the binaries because it will bloat the archive size for users that don't want the AppImage.

Reply:
Might make sense to do this as a separate workflow altogether?

Reply:
Do you mean the workflow artifacts, or the release assets? I'm pretty sure none of the other release assets contain the AppImage, just the AppImage itself (see [my test release here](https://github.com/mtoohey31/helix/releases/tag/22.45) vs the sizes of [the latest official release](https://github.com/helix-editor/helix/releases/tag/22.03)), though the AppImage does get bundled into the linux release artifacts if that's a problem.

Reply:
> though the AppImage does get bundled into the linux release artifacts if that's a problem.

Yes, that's what I meant. Various distributions simply repackage the linux release, bundling AppImage into the same archive will increase the download size. The AppImage is a separate package/distribution format and it should be packaged into it's own archive.

Reply:
Ah shoot, I meant to say _workflow_ artifacts. Just to make sure we're on the same page, the AppImage gets bundled into the linux artifacts saved [here in the workflow run](https://github.com/mtoohey31/helix/actions/runs/2168263969), but [_not here_ on the release page](https://github.com/mtoohey31/helix/releases/tag/22.45). Are distributions using the workflow artifacts instead of the release artifacts?

Reply:
Oh I see. I think this is currently fine: the AppImage is built in the build step for the x86_64 linux image and it also ends up in the workflow artifacts but then it ends up as a separate release artifact.